jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -422,7 +422,7 @@ ecma_copy_ecma_string (ecma_string_t *string_desc_p) /**< string descriptor */
       ecma_string_heap_header_t *new_data_p = (ecma_string_heap_header_t *) mem_heap_alloc_block (data_size);
       memcpy (new_data_p, data_p, data_p->size + sizeof (ecma_string_heap_header_t));
 
-      ECMA_SET_NON_NULL_POINTER (new_str_p->u.collection_cp, data_p);
+      ECMA_SET_NON_NULL_POINTER (new_str_p->u.collection_cp, new_data_p);
```

Comment:
Here the problem is that there is a reference to the old string. A regression test would look like this:

    var a = "string";
    var b = [];
    for (i =0 i < 10000; i++) b[i] = a;
    gc();
    for (i =0 i < 9000; i++) b[i] = null;
    gc(); // could be a crash